docs/tutorials/forecasting/hierarchical_model_tutorial.md.template

```diff
@@ -0,0 +1,177 @@
+aggregation matrix.
+`GluonTS` provides a simple way to construct hierarchical time series by reading bottom-level
+time series and the aggregation matrix from csv files.
+Here we first describe the preparation of hierarchical time series before discussing the
+training of the hierarchical model.
+
+
+## Preparation of Hierarchical Time Series
+
+The bottom level time series are assumed to be available in a csv file as columns.
+The csv file should also contain the index column listing the time stamps for each row.
+Note that the aggregated time series are automatically constructed and should not be provided.
+Here is an [example csv](https://gist.githubusercontent.com/rshyamsundar/39e57075743537c4100a716a7b7ec047/raw/f02f5aeadad73e3f3e9cf54478606d3507826492/example_bottom_ts.csv) of bottom level time series.
+Similarly, the aggregation matrix can also be read from a csv file;
+here is an [example](https://gist.githubusercontent.com/rshyamsundar/17084fd1f28021867bcf6f2d69d9b73a/raw/32780ca43f57a78f2d521a75e73b136b17f34a02/example_agg_mat.csv).
+Note that grouped time series can also be represented by an aggregation matrix in the same
+way as the hierarchical time series and hence the material presented here is also applicable to
+grouped time series.
+
+
+```python
+import pandas as pd
+from gluonts.dataset.hierarchical import HierarchicalTimeSeries
+
+# Load (only!) the time series at the bottom level of the hierarchy.
+ts_at_bottom_level_csv = (
+    "https://gist.githubusercontent.com/rshyamsundar/39e57075743537c4100a716a7b7ec047/"
+    "raw/f02f5aeadad73e3f3e9cf54478606d3507826492/example_bottom_ts.csv"
+)
+ts_at_bottom_level = pd.read_csv(
+    ts_at_bottom_level_csv,
+    index_col=0,
+    parse_dates=True,
+)
+
+# Make sure the dataframe has `PeriodIndex`.
+ts_at_bottom_level = ts_at_bottom_level.to_period()
+
+# Load the aggregation matrix `S`.
+S_csv = (
+    "https://gist.githubusercontent.com/rshyamsundar/17084fd1f28021867bcf6f2d69d9b73a/raw/"
+    "32780ca43f57a78f2d521a75e73b136b17f34a02/example_agg_mat.csv"
+)
+S = pd.read_csv(S_csv).values
+
+# Create `HierarchicalTimeSeries` object.
+hts = HierarchicalTimeSeries(
+    ts_at_bottom_level=ts_at_bottom_level,
+    S=S,
+)
+
+# You can access all the time series of the hierarchy using `ts_at_all_levels` property.
+hts.ts_at_all_levels.head()
+```
+
+## Use case 1 - Simple Model Training and Forecasting
+
+We now show the simplest use-case where we want to train the model on the whole dataset available
+and produce predictions for the future time steps.
+We assume that the hierarchical time series `hts` of type `HierarchicalTimeSeries` has already been
+constructed as described above.
+The first step is to convert this hierarchical time series to a `Dataset` on which
+mini-batch training can be run.
+Here we convert it into `gluonts.dataset.pandas.PandasDataset`.
+
+```python
+from gluonts.dataset.hierarchical import to_pandas_dataset
+
+dataset = to_pandas_dataset(hts=hts)
```

Comment:
This could be a method of `HierarchicalTimeSeries`:

```python
hts.to_pandas_dataset()
```

I'm wondering whether `pandas` should be in the name here though: in the end, what matters to the user is that this is converting a `HierarchicalTimeSeries` object into a `Dataset`, so maybe `to_dataset` is a better name.

Reply:
ya, actually it is better it if it a method of `HierarchicalTimeSeries`.

I tried being explicit about to which dataset I am converting to (`PandasDataset`). `to_dataset` is also fine and we don't imagine converting this to to other types of datasets anyway right?

Reply:
Now I remember why I moved `to_pandas_dataset` out of the class. We are not converting just the `hts` to `Dataset` here. We are taking external features and the target time series (happened to be `hts`) to create a dataset. So I thought `to_dataset(hts, features)` is better than `hts.to_dataset(features)`. Your call!

Reply:
Right, but features are optional, so I’d say making it a method is fine

Reply:
Moved.